src/libraries/System.Transactions.Local/src/System/Transactions/Oletx/OletxTransaction.cs

```diff
@@ -0,0 +1,1395 @@
+        // we need to remember that transaction because GetRealObject is called twice during
+        // deserialization.  In this case, GetRealObject returns the LTM transaction, not this OletxTransaction.
+        // The OletxTransaction will get GC'd because there will be no references to it.
+        internal Transaction? SavedLtmPromotedTransaction;
+
+        private TransactionTraceIdentifier _traceIdentifier = TransactionTraceIdentifier.Empty;
+
+        // Property
+        internal RealOletxTransaction RealTransaction
+            => RealOletxTransaction;
+
+        internal Guid Identifier
+        {
+            get
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceOleTx, this, $"{nameof(OletxTransaction)}.{nameof(Identifier)}");
+                }
+
+                Guid returnValue = RealOletxTransaction.Identifier;
+
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodExit(TraceSourceType.TraceSourceOleTx, this, $"{nameof(OletxTransaction)}.{nameof(Identifier)}");
+                }
+
+                return returnValue;
+            }
+        }
+
+        internal Guid DistributedTxId
+        {
+            get
+            {
+                Guid returnValue = Guid.Empty;
+
+                if (RealOletxTransaction != null && RealOletxTransaction.InternalTransaction != null)
+                {
+                    returnValue = RealOletxTransaction.InternalTransaction.DistributedTxId;
+                }
+
+                return returnValue;
+            }
+        }
+
+        internal TransactionStatus Status
+        {
+            get
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceOleTx, this, $"{nameof(OletxTransaction)}.{nameof(Status)}");
+                }
+
+                TransactionStatus returnValue = RealOletxTransaction.Status;
+
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodExit(TraceSourceType.TraceSourceOleTx, this, $"{nameof(OletxTransaction)}.{nameof(Status)}");
+                }
+
+                return returnValue;
+            }
+        }
+
+        internal Exception? InnerException
+            => RealOletxTransaction.InnerException;
+
+        internal OletxTransaction(RealOletxTransaction realOletxTransaction)
+        {
+            RealOletxTransaction = realOletxTransaction;
+
+            // Tell the realOletxTransaction that we are here.
+            RealOletxTransaction.OletxTransactionCreated();
+        }
+
+        // TODO: Do we want to keep supporting this?
+        protected OletxTransaction(SerializationInfo? serializationInfo, StreamingContext context)
```

Comment:
Note this serialization constructor.

I'm not sure what the state of serialization is exactly - this doesn't use BinaryFormatter, just serializes as an opaque byte array.

/cc @stephentoub @GrabYourPitchforks

Reply:
> // TODO: Do we want to keep supporting this?

I think "no" :-) but @GrabYourPitchforks should make the call.